docs/src/modules/spring/pages/serialization.adoc

```diff
@@ -1 +1,38 @@
 = Handling Serialization
+
+You do not need to handle serialization for the messages of an Event Source Entity or a Value Entity. On the other hand, Kalix exposes endpoints with JSON messages and you need to handle serialization of their inputs and outputs: if you are using Java classes you need to annotate them with the https://github.com/FasterXML/jackson-annotations#usage-general[proper Jackson annotation] but if you are using Java https://openjdk.org/jeps/395[record] no annotation is needed. 
```

Comment:
```suggestion
You do not need to handle serialization for the messages of an Event Source Entity or a Value Entity. On the other hand, Kalix exposes endpoints with JSON messages and you need to handle serialization of their inputs and outputs: if you are using Java classes you need to annotate them with the https://github.com/FasterXML/jackson-annotations#usage-general[proper Jackson annotation] but if you are using Java https://openjdk.org/jeps/395[record] no annotation is needed. 
```

Here we give the impression that only input and output matters. But events and state are also serializable. Users don't need to serialize and deserialize them themselves, but they do need to create classes that can be serializable. Either by using records or by annotating their domains with Jackson.

Reply:
changed now